### PR TITLE
Bevm -- Ethereum Virtual Machine integration in ByzCoin

### PR DESCRIPTION
**Pull request by cgrigis**
This introduces the BEvm contract, allowing to execute Ethereum smart contracts in ByzCoin. The `contracts` subdirectory contains some example and test smart contracts, some of them written in Scala and transpiled by the [Stainless](https://github.com/epfl-lara/smart) formal verification tool from LARA.
See README for more details.

All comments welcome!

### CODE REVIEW

File: bevm/bevm_client.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ type EvmAccount struct {
 func NewEvmAccount(privateKey string) (*EvmAccount, error) {
 	privKey, err := crypto.HexToECDSA(privateKey)
 	if err != nil {
-		return nil, err
+		return nil, errors.New("Error decoding private key: " + err.Error())
```
Comment (ineiti) [inline bevm/bevm_client.go]: I'm surprised that `golint` doesn't complain about error-message starting with a capital letter. I remember it being quite insistant on that one.

Comment (cgrigis) [reply bevm/bevm_client.go]: I am assuming you are referring to [this](https://github.com/golang/go/wiki/CodeReviewComments#error-strings), thank you I was not aware of this guideline, will fix.

**Comment by Gilthoniel** [comment]
Just so you know, we're waiting for a 3.2.1 release to review and merge as 3.3.x.

**Comment by ineiti** [comment]
> Just so you know, we're waiting for a 3.2.1 release to review and merge as 3.3.x.

Does that mean that you want to release new conodes with a 3.2.1 version, and then merge the BEVM code into the 3.3?

**Comment by Gilthoniel** [comment]
Yes exactly

**Comment by cgrigis** [comment]
> Just so you know, we're waiting for a 3.2.1 release to review and merge as 3.3.x.

OK, thank you for the update.

**Comment by jeffallen** [comment]
I'm now actively working on this.

**Comment by ineiti** [comment]
With regard to the cross-compilation, if I understand correctly, the main blockage is that it's not possible to do anymore `make bindist` in the `conode`-directory.

One way to get around this would be to have automated builds with travis, which would allow to use darwin-dockers, windows-dockers, and linux-dockers, to create the corresponding binary-builds. Then there would also be up-to-date builds for every merge: https://twitter.com/mipsytipsy/status/1178250438354685953

**Pull request merged**